### PR TITLE
Codechange: rename SettingGuiFlag to SettingFlag 

### PR DESCRIPTION
**Pull request by TrueBrain**
## Motivation / Problem

@rubidium42 : it happened, it annoyed me enough!!

For the rest: settings have this weird quirk where something is called "guiflags" but is really the flags of the settings. And we have a "flags" which is really the flags of the saveload part. This is so confusing ...

## Description

```
It is a lovely organicly grown enum, where it started off with
GUI-only flags, and after that a few flags got added that can be
considered GUI-only (the GUI disables/enables based on them), to
only have flags added that has nothing to do with the GUI.

So be less confusing, and rename them to what they do.

Additionally, I took this opportunity to rename 0ISDISABLED to
reflect what it really does.
```

In a follow-up commit I am going to address the `flags` vs `guiflags`.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/settings_internal.h [hunk 1]
```diff
@@ -12,21 +12,20 @@
...
+	SF_GUI_0_IS_SPECIAL  = 1 << 0, ///< a value of zero is possible and has a custom string.
+	SF_GUI_DISPLAY_ABS   = 1 << 1, ///< display absolute value of the setting.
+	SF_GUI_MULTISTRING   = 1 << 2, ///< the value represents a limited number of string-options (internally integer).
+	SF_GUI_CURRENCY      = 1 << 3, ///< the number represents money, so when reading value multiply by exchange rate.
+	SF_NETWORK_ONLY      = 1 << 4, ///< this setting only applies to network games.
+	SF_NO_NETWORK        = 1 << 5, ///< this setting does not apply to network games; it may not be changed during the game.
+	SF_NEWGAME_ONLY      = 1 << 6, ///< this setting cannot be changed in a game.
+	SF_SCENEDIT_TOO      = 1 << 7, ///< this setting can be changed in the scenario editor (only makes sense when SF_NEWGAME_ONLY is set).
+	SF_SCENEDIT_ONLY     = 1 << 8, ///< this setting can only be changed in the scenario editor.
+	SF_PER_COMPANY       = 1 << 9, ///< this setting can be different for each company (saved in company struct).
```
Comment (LordAro) [inline src/settings_internal.h]: Is reordering this going to cause any issues?

Comment (TrueBrain) [reply src/settings_internal.h]: not as far as I can see and as far as I could test.

Nowhere this value is stored in the savegame. But I would love some extra pairs of eyes on it to validate that claim.

File: src/settings_internal.h [hunk 2]
```diff
@@ -12,21 +12,20 @@
...
-	SGF_CURRENCY      = 1 << 4, ///< the number represents money, so when reading value multiply by exchange rate
-	SGF_NO_NETWORK    = 1 << 5, ///< this setting does not apply to network games; it may not be changed during the game
-	SGF_NEWGAME_ONLY  = 1 << 6, ///< this setting cannot be changed in a game
-	SGF_SCENEDIT_TOO  = 1 << 7, ///< this setting can be changed in the scenario editor (only makes sense when SGF_NEWGAME_ONLY is set)
-	SGF_PER_COMPANY   = 1 << 8, ///< this setting can be different for each company (saved in company struct)
-	SGF_SCENEDIT_ONLY = 1 << 9, ///< this setting can only be changed in the scenario editor
+enum SettingFlag : uint16 {
+	SF_NONE = 0,
+	SF_GUI_0_IS_SPECIAL  = 1 << 0, ///< a value of zero is possible and has a custom string.
+	SF_GUI_DISPLAY_ABS   = 1 << 1, ///< display absolute value of the setting.
```
Comment (frosch123) [inline src/settings_internal.h]: Add "positive and negative values use different strings".
To match the previous enum, maybe "SF_GUI_SIGN_IS_SPECIAL" :p

File: src/settings_internal.h [hunk 3]
```diff
@@ -12,21 +12,20 @@
...
+	SF_GUI_MULTISTRING   = 1 << 2, ///< the value represents a limited number of string-options (internally integer).
```
Comment (frosch123) [inline src/settings_internal.h]: SF_GUI_DROPDOWN ?

File: src/table/settings/company_settings.ini [hunk 4]
```diff
@@ -16,15 +16,15 @@ static const SettingTable _company_settings{
...
+SDT_BOOL = SDT_BOOL(CompanySettings, $var,        $saveflags, $flags, $def,                        $str, $strhelp, $strval, $pre_cb, $post_cb, $from, $to, $cat, $extra, $startup),
+SDT_VAR  =  SDT_VAR(CompanySettings, $var, $type, $saveflags, $flags, $def, $min, $max, $interval, $str, $strhelp, $strval, $pre_cb, $post_cb, $from, $to, $cat, $extra, $startup),
 
 [validation]
 SDT_VAR = static_assert($max <= MAX_$type, "Maximum value for CompanySettings.$var exceeds storage size");
 
 [defaults]
 saveflags= 0
-guiflags = SF_PER_COMPANY
+flags    = SF_PER_COMPANY
```
Comment (frosch123) [inline src/table/settings/company_settings.ini]: I think this three-way-rename "guiflags" -> "flags" -> "saveflags" is unfair for rebasing patches/forks.
How about "sf"?

File: src/settings_internal.h (hunk 1 above)
Comment (rubidium42) [reply src/settings_internal.h]: Shouldn't the comments start with a capital?

File: src/settings_internal.h (hunk 2 above)
Comment (frosch123) [reply src/settings_internal.h]: Or SF_GUI_NEGATIVE_IS_SPECIAL

**Review by frosch123 — DISMISSED** [review]

**Review by frosch123 — APPROVED** [review]

**Pull request merged**